pkg/client/powervs_client.go

```diff
@@ -526,9 +526,13 @@ func resolveEndpoints(ctrlRuntimeClient client.Client) (map[string]string, error
 }
 
 func setCustomEndpoints(customEndpointsMap map[string]string, keys []string) error {
-	//TODO(Doubt): Is it required to validate the endpoints again before setting it or blindly trust installer
 	for _, key := range keys {
+		if _, ok := endPointKeyToEnvNameMap[key]; !ok {
+			klog.Infof("setCustomEndpoints ignoring %s", key)
+			continue
+		}
 		if val, ok := customEndpointsMap[key]; ok {
+			fmt.Printf("setCustomEndpoints: key = %s, value = %s", key, val)
```

Comment:
Lets use klog.Infof instead of fmt.Printf

Reply:
I will remove this as it was a holdover